test/config/utility.cc

```diff
@@ -671,6 +671,9 @@ void ConfigHelper::setConnectConfig(
 
   hcm.add_upgrade_configs()->set_upgrade_type("CONNECT");
   hcm.mutable_http2_protocol_options()->set_allow_connect(true);
+  if (http3) {
+    hcm.mutable_http3_protocol_options()->set_allow_upgrade_connect(true);
```

Comment:
I think the H2 option is (sadly) misnamed because HTTP/2 allows standard connect without a special knob, and just disallows the upgrade style connect.